### PR TITLE
feat(http): add transfer cache module from nguniversal/common

### PR DESCRIPTION
**Pull request by CaerusKaru**
* Adds TransferHttpCacheModule from nguniversal/common to avoid
  duplicate HTTP calls between server and browser
* Bump RxJS dep to 5.5.5 to avoid issues with lower patch versions
* Clean up certain areas of code, remove redundant imports

PR close #509

### CODE REVIEW

**Comment by CaerusKaru** [comment]
cc @vikerman @Toxicable

**Comment by Toxicable** [comment]
Looks good aside from the linting changes, revert those for now.
If you want to fix that up another PR adding a tslint would be great, the default one produce by `ng new` will do

**Comment by CaerusKaru** [comment]
@Toxicable on it!

**Comment by CaerusKaru** [comment]
#511 created to deal with linting

**Comment by m98** [comment]
Nice job :+1: 

I think it would be good if you can provide an example of it too
Maybe a new component for HTTP request representing and how does cache work.

**Comment by CaerusKaru** [comment]
@Toxicable thoughts on @m98's comments? Personally I don't see the usefulness of giving an example for `TransferHttpCache` since it's pretty self-explanatory, and is mostly here to give a leg-up to developers getting started. But I'm open to adding something if you think it would be useful.

@m98 I still think it would be invaluable if you could provide another example to the repo for demonstrating usage of the `TransferState` feature, just without HTTP.

**Comment by m98** [comment]
I did not mean adding an example to demonstrating usage of the  `TransferState`

But since this pull request adds ability to transfer state from server to browser, it might be helpful to use a simple HTTP request just let users know there is already something which is supporting transfer state.

They might write their own services to support this if they don't know such thing added to the repo

**Comment by Toxicable** [comment]
I think adding an example for each would be the better path for this one, so that we can compare and show the different situations you'd use either

File: .gitignore [hunk 1]
```diff
@@ -2,6 +2,7 @@
 .DS_Store
 .vscode
 morgan.log
+package-lock.json
```
Comment (Toxicable) [inline .gitignore]: uhhhh we shouldn't be ignoring this file.
We should either use Yarn or Npm and stick to just one.
I vote Yarn :P

File: package.json [hunk 2]
```diff
@@ -39,10 +39,11 @@
     "@angular/platform-browser-dynamic": "^5.0.0",
     "@angular/platform-server": "^5.0.0",
     "@angular/router": "^5.0.0",
+    "@nguniversal/common": "^5.0.0-beta.5",
     "@nguniversal/express-engine": "^5.0.0-beta.5",
     "@nguniversal/module-map-ngfactory-loader": "^5.0.0-beta.5",
     "core-js": "^2.4.1",
-    "rxjs": "^5.5.2",
+    "rxjs": "^5.5.5",
```
Comment (Toxicable) [inline package.json]: this change is unneeded

Comment (CaerusKaru) [reply package.json]: Fixed

File: .gitignore (hunk 1 above)
Comment (CaerusKaru) [reply .gitignore]: Fixed

**Pull request merged**